### PR TITLE
Update SQA reports allowances to avoid regression

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #12049 

This will prevent regression like in the Navier Stokes module for example, where an old FE NS implementation was merged with insufficient doco. 1 year later, it has not been contributed (actually I think this particular instance would still go through sadly).

I am also turning back on colors in the SQA reports. The modules are pretty close now, that orange can just serve as extra motivation. The worst one (THM) got fixed.

### CODE REVIEW

File: modules/doc/sqa_reports.yml [hunk 1]
```diff
@@ -485,7 +445,6 @@ Requirements:
         log_missing_issues: WARNING
         log_testable: WARNING
         log_duplicate_requirement: WARNING
-        show_warning: false
```
Comment (GiudGiud) [inline modules/doc/sqa_reports.yml:488]: the items above will have to wait for #26359 to get in

**Review by lindsayad — COMMENTED** [review]
The concept here sounds good to me, but I'm going to leave this review to @cticenhour

**Comment by cticenhour** [comment]
Thanks @lindsayad. @GiudGiud are you done pushing to the PR? If so, I should be able to look at this sometime today between meetings.

File: modules/misc/doc/content/source/kernels/CoefDiffusion.md [hunk 2]
```diff
@@ -1 +1,14 @@
-!template load file=stubs/moose_object.md.template name=CoefDiffusion syntax=/Kernels/CoefDiffusion
\ No newline at end of file
+# CoefDiffusion
```
Comment (cticenhour) [inline modules/misc/doc/content/source/kernels/CoefDiffusion.md:1]: Thanks for moving these off the old template!

File: modules/misc/doc/content/source/kernels/ThermoDiffusion.md [hunk 3]
```diff
@@ -1 +1,24 @@
...
+
+`ThermoDiffusion` implements thermodiffusion (also called thermophoresis, thermomigration, or the Soret effect)
+or the movement of species due a temperature gradient. The mass flux $J$
+is given as
+
+\begin{equation}
+\mathbf{J}=-S\nabla T
+\end{equation}
+
+where $S$ is Soret diffusion coefficient which is typically a combination of the
```
Comment (cticenhour) [inline modules/misc/doc/content/source/kernels/ThermoDiffusion.md:13]:
```suggestion
where $S$ is Soret diffusion coefficient, which is typically a combination of the
```

**Comment by GiudGiud** [comment]
yup done. Had to tackle one last little thing that s it

**Review by cticenhour — CHANGES_REQUESTED** [review]
Small tweaks to the new written doco, but otherwise good.

**Comment by GiudGiud** [comment]
done

**Review by cticenhour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**